framework/src/modules/chain/init_steps/create_socket_cluster.js

```diff
@@ -0,0 +1,82 @@
+const path = require('path');
+const SocketCluster = require('socketcluster');
+const promisifyEvent = require('p-event');
+const MasterWAMPServer = require('wamp-socket-cluster/MasterWAMPServer');
+const wsRPC = require('../api/ws/rpc/ws_rpc').wsRPC;
+const WsTransport = require('../api/ws/transport');
+
+const workersControllerPath = path.join(__dirname, '../workers_controller');
+
+module.exports = async ({
+	config,
+	network,
+	modules: { transport },
+	components: { logger },
+}) => {
+	if (!config.peers.enabled) {
+		logger.info(
+			'Skipping P2P server initialization due to the config settings - "peers.enabled" is set to false.'
+		);
+		return true;
+	}
+
+	const webSocketConfig = {
+		workers: 1,
+		port: config.wsPort,
+		host: '0.0.0.0',
+		wsEngine: config.peers.options.wsEngine,
+		workerController: workersControllerPath,
+		perMessageDeflate: false,
+		secretKey: 'liskSecretKey',
+		// Because our node is constantly sending messages, we don't
+		// need to use the ping feature to detect bad connections.
+		pingTimeoutDisabled: true,
+		// Maximum amount of milliseconds to wait before force-killing
+		// a process after it was passed a 'SIGTERM' or 'SIGUSR2' signal
+		processTermTimeout: 10000,
+		logLevel: 0,
+	};
+
+	const childProcessOptions = {
+		version: config.version,
+		minVersion: config.minVersion,
+		protocolVersion: config.protocolVersion,
+		nethash: config.nethash,
+		port: config.wsPort,
+		nonce: config.nonce,
+		blackListedPeers: config.peers.access.blackList,
+	};
+
+	const socketCluster = new SocketCluster(webSocketConfig);
+	network.app.rpc = wsRPC.setServer(
+		new MasterWAMPServer(socketCluster, childProcessOptions)
+	);
+
+	// The 'fail' event aggregates errors from all SocketCluster processes.
+	socketCluster.on('fail', err => {
+		logger.error(err);
+		if (err.name === 'WSEngineInitError') {
+			const extendedError = logger.error(extendedError);
+		}
+	});
+
+	socketCluster.on('workerExit', workerInfo => {
+		let exitMessage = `Worker with pid ${workerInfo.pid} exited`;
+		if (workerInfo.signal) {
+			exitMessage += ` due to signal: '${workerInfo.signal}'`;
+		}
+		logger.error(exitMessage);
+	});
+
+	return promisifyEvent(socketCluster, 'ready').then(() => {
```

Comment:
Anything else we could use here not to introduce the new dependency to `p-event`? We introduced several with 1.6 anyway.

Reply:
I also wish not to, but in one review it was suggested to promisify event. I could not find any native utility method for it. Suggest either we create our own `helper` :) for it or don't promisify events.

Reply:
maybe we can do
```
await new Promise(resolve => {
  socketCluster.on('ready', resolve);
});
logger.info()
...
```
@nazarhussain  please consult with @jondubois

Reply:
@shuse2 @MaciejBaj I am well aware of using Promise constructor ;) But if you look at this issue https://github.com/LiskHQ/lisk-modular/issues/51 it was suggested to not use Promise constructor instead use some library to promisify events.

Reply:
After the discussion with @MaciejBaj we decided not to promisify event handlers. We will stick to the NodeJS event handler approach everywhere.

Reply:
I agree that it's better `event` and `Promise` is different